fvm/evm/handler/handler.go

```diff
@@ -547,12 +552,10 @@ func (h *ContractHandler) emitEvent(event *events.Event) error {
 	return h.backend.EmitEvent(ev)
 }
 
-func (h *ContractHandler) getBlockContext() (types.BlockContext, error) {
-	bp, err := h.blockStore.BlockProposal()
```

Comment:
`getBlockContext()` used to always perform a storage read operation, for fetching the latest block proposal.

However, some of the methods that use `getBlockContext()`, would also read the latest block proposal from storage, since it was necessary for their functionality.

To avoid duplicated storage read operations, we require callers of `getBlockContext()`, to pass in the latest block proposal.